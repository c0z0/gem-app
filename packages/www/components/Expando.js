import React, { useEffect, useRef, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  height: ${({ open, openHeight }) => (open ? openHeight : '0px')};
  opacity: ${({ open }) => (open ? '1' : '0')};
  transition: all 0.2s;
  overflow: ${({ open }) => (open ? 'inherit' : 'hidden')};
  @media (${({ theme }) => theme.b.tabletUp}) {
    padding-top: ${({ open }) => (open ? '12px' : '0')};
  }
`

export default function Expando({ children, open }) {
  const contentRef = useRef(null)
  const [heightState, setHeight] = useState('0px')

  useEffect(() => {
    const height = contentRef.current.offsetHeight

    setHeight(`${height}px`)
  }, [children])

  return (
    <Wrapper open={open} openHeight={heightState}>
      <div ref={contentRef}>{children}</div>
    </Wrapper>
  )
}

Expando.defaultProps = {
  children: null
}

Expando.propTypes = {
  open: PropTypes.bool.isRequired,
  children: PropTypes.node
}
