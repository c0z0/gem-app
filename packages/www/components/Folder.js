import React, { useEffect, useState, useRef } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

function outerHeight(el) {
  let height = el.offsetHeight
  const style = getComputedStyle(el)

  height += parseInt(style.marginTop, 10) + parseInt(style.marginBottom, 10)
  return height
}

const Wrapper = styled.div`
  height: ${({ open, openHeight }) => (open ? openHeight : '0px')};
  opacity: ${({ open }) => (open ? '1' : '0')};
  transition: all 0.2s;
  overflow: ${({ open }) => (open ? 'inherit' : 'hidden')};
  @media (${({ theme }) => theme.b.tabletUp}) {
    padding-top: ${({ open }) => (open ? '12px' : '0')};
  }
`

const Carret = styled.img.attrs({ src: '/static/carret.svg' })`
  transform: rotate(${({ flipped }) => (flipped ? '180deg' : '0deg')});

  transition: transform 0.2s;
  margin-right: 10px;
`

const Content = styled.div`
  padding-left: 32px;
`

const FolderWrapper = styled.div`
  &:not(:last-child) {
    border-bottom: 1px solid #eee;
    margin-bottom: 28px;
  }
`

const Title = styled.h4.attrs({ role: 'button' })`
  font-weight: normal;
  padding: 16px 0;
  font-size: 20px;
  cursor: pointer;
  margin: 0;
  display: flex;
  align-items: center;
`

export default function Folder({
  children,
  title,
  onDrop,
  onDragOver,
  onDragLeave
}) {
  const contentRef = useRef(null)
  const [heightState, setHeight] = useState('0px')
  const [visibleState, setVisibleState] = useState(false)

  useEffect(() => {
    const height = outerHeight(contentRef.current)

    setHeight(`${height}px`)
  }, [children])

  return (
    <FolderWrapper
      onDrop={onDrop}
      onDragOver={onDragOver}
      onDragLeave={onDragLeave}
    >
      <Title onClick={() => setVisibleState(!visibleState)}>
        <Carret flipped={visibleState} />
        <span>{title}</span>
      </Title>
      <Wrapper open={visibleState} openHeight={heightState}>
        <Content ref={contentRef}>{children}</Content>
      </Wrapper>
    </FolderWrapper>
  )
}

Folder.defaultProps = {
  children: null,
  onDrop: () => {},
  onDragOver: () => {},
  onDragLeave: () => {}
}

Folder.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string.isRequired,
  onDrop: PropTypes.func,
  onDragOver: PropTypes.func,
  onDragLeave: PropTypes.func
}
