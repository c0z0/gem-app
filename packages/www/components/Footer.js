import React from 'react'
import styled from 'styled-components'

import Container from './Container'
import { Diamond } from './Svg'

const FooterContainer = styled(Container)`
  display: flex;
  justify-content: space-between;
  align-items: center;

  @media (${({ theme }) => theme.b.phoneOnly}) {
    flex-direction: column;
  }
`

const Border = styled.footer`
  background: #fafafa;
  border-top: solid 1px #eaeaea;
  padding: 48px 0;
  color: #cecece;
  position: absolute;
  right: 0;
  bottom: 0;
  left: 0;
`

const Logo = styled.div`
  display: flex;
  align-items: center;
`

const Title = styled.span`
  margin-left: 8px;
  font-weight: 400;
  margin-right: 20px;
  color: #cecece;
  font-size: 20px;
`

const StyledDiamond = styled(Diamond).attrs({
  alt: 'gem logo'
})`
  width: 40px;
  filter: grayscale(100%);
  opacity: 0.5;
`

const Link = styled.a`
  color: #cecece;
  text-decoration: none;
  transition: all 0.2s;

  &:hover {
    color: #484848;
  }
`

export default function Footer() {
  return (
    <Border>
      <FooterContainer>
        <Logo>
          <StyledDiamond />
          <Title>Gem</Title>
        </Logo>
        <p>
          Made by <Link href="https://cserdean.com">cserdean.com.</Link>
        </p>
      </FooterContainer>
    </Border>
  )
}
