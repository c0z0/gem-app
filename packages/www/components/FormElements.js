import styled from 'styled-components'

export const Input = styled.input`
  width: 275px;
  max-width: 100%;
  padding: 8px;
  font-size: 16px;
  outline: none;
  color: #484848;
  border: 1px solid #ddd;
  border-radius: 6px;
  transition: all 0.2s;
  margin: 1rem 0;

  &::placeholder {
    color: #ddd;
  }

  &:focus {
    border-color: #aaa;
  }

  ${({disabled}) =>
    disabled &&
    `border-color: #ddd; color: #ddd; background: #fafafa; cursor: not-allowed;`}
`

export const Button = styled.button`
  outline: none;
  border: none;
  border-radius: 6px;
  padding: 8px 12px;
  font-size: 16px;
  margin: 4px;
  color: white;
  cursor: pointer;
  background: ${({theme}) => theme.main};
  border: 1px solid transparent;

  ${({disabled}) =>
    disabled &&
    `color: #ddd; border-color: #ddd; background: #fafafa; cursor: not-allowed;`}
`
