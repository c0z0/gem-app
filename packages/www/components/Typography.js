import React from 'react'
import styled, { keyframes } from 'styled-components'

export const Title = styled.h1`
  font-weight: 400;
  color: ${({ theme }) => theme.main};
  font-size: 24px;
  ${({ light }) => light && `color: white;`}
  margin: 0;
`

export const Card = styled.div`
  padding: 48px;
  border-radius: 7px;
  background: white;
  box-shadow: 0px 4px 30px rgba(0, 0, 0, 0.15);

  @media (${({ theme }) => theme.b.phoneOnly}) {
    margin: 0;
    box-shadow: none;
    padding-top: 32px;
  }
`

export const SubTitle = styled.h2`
  font-weight: 400;
  text-transform: lowercase;
  font-size: 24px;
  color: #98a4a8;
  ${({ light }) => light && `color: white;`}
  margin: 0;
  margin-bottom: 24px;
`

export const P = styled.p`
  line-height: 1.5rem;
  ${({ light }) => light && `color: white;`}
`

export const H1 = styled.h1`
  font-size: 32px;
  margin: 2em 0 1em 0;
  font-weight: 600;

  @media (${({ theme }) => theme.b.phoneOnly}) {
    margin: 1em 0 1em 0;
  }
`

const ElipsisAnimation = keyframes`
  0% {
    opacity: 0.2;
  }
  20% { 
    opacity: 1;
  }
  100% {
    opacity: 0.2;
  }
`

const Elipsis = styled.span`
  animation: ${ElipsisAnimation} 1.4s infinite;

  &:nth-child(2) {
    animation-delay: 0.2s;
  }
  &:nth-child(3) {
    animation-delay: 0.4s;
  }
`

export function LoadingElipsis() {
  return (
    <React.Fragment>
      <Elipsis key="a">.</Elipsis>
      <Elipsis key="b">.</Elipsis>
      <Elipsis key="c">.</Elipsis>
    </React.Fragment>
  )
}
