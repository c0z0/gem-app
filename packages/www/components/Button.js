import styled from 'styled-components'

import { LoadingElipsis } from './Typography'

const Button = styled.button.attrs({ role: 'button' })`
  font-size: 16px;
  color: white;
  background: ${({ theme }) => theme.main};

  padding: 0 2rem;
  line-height: 2.5rem;
  margin: 0;
  border-radius: 7px;
  height: 2.5rem;
  box-shadow: 0px 4px 14px rgba(117, 72, 155, 0.39);
  transition: all 0.2s;
  outline: none;
  border: none;
  cursor: pointer;

  text-transform: capitalize;

  &:hover {
    opacity: 0.9;
    box-shadow: 0px 6px 20px rgba(117, 72, 155, 0.39);
  }

  ${({ secondary, theme }) =>
    secondary &&
    `
    color: ${theme.main};
    background: white;
    box-shadow: 0px 4px 14px rgba(0, 0, 0, 0.1);

    &:hover {
      box-shadow: 0px 6px 20px rgba(0, 0, 0, 0.12);
    }

  `}

${({ small }) =>
  small &&
  `
    height: 2rem;
    line-height: 2rem;

  `}

${({ flat }) =>
  flat &&
  `
    box-shadow: none;
    &:hover {
      box-shadow: none;
    }
  `}

  ${({ loading, disabled }) =>
    (loading || disabled) &&
    `
    cursor: not-allowed;
    color: #CCCCCC;
    box-shadow: none !important;
    opacity: 1 !important;
    background: #EFEFEF;
  `}

`

Button.Elipsis = LoadingElipsis

export default Button
