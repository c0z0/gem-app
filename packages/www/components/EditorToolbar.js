import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Link from 'next/link'

import SubMenu from './SubMenu'
import * as Icons from './Svg'
import { LoadingElipsis } from './Typography'

const Action = styled.button`
  outline: none;
  background: white;
  z-index: 2;
  cursor: pointer;
  padding: 8px 10px;
  font-size: 21px;
  transition: all 0.2s;
  color: #484848;

  display: flex;
  align-items: center;

  &:hover {
    background: #fafafa;
    color: #484848;
  }

  ${({ active }) =>
    active &&
    ` background: red;
    `}

  color: ${({ theme }) => theme.text};
  font-weight: 500;
  border: 1px solid #eaeaea;
  border-right: none;

  &:first-child {
    border-radius: 4px 0 0 4px;
    border-left: 1px solid #eaeaea;
  }

  &:last-child {
    border-radius: 0 4px 4px 0;
    border-right: 1px solid #eaeaea;
  }

  &:only-child {
    border-radius: 4px;
  }


  ${({ divider }) => divider && `border-right: 1px solid #ddd;`}
`

const SaveStatus = styled.span`
  color: #ddd;
  align-self: center;
  margin-left: 8px;
  @media (${({ theme }) => theme.b.phoneOnly}) {
    flex: 1;
    text-align: right;
  }
`

const Controls = styled.div`
  display: flex;
`

const Carret = styled.img.attrs({ src: '/static/carret.svg' })`
  transform: rotate(90deg);
  margin-right: 8px;
  color: #484848;
`

const BackButton = styled.a`
  align-self: center;
  margin-right: 8px;
  display: flex;
  color: #aaa;
  text-decoration: none;
  align-items: center;
  transition: all 0.2s;

  @media (${({ theme }) => theme.b.phoneOnly}) {
    flex: 1;
  }

  :hover {
    color: #484848;
  }
`

export default function EditorToolbar({
  onToggleMark,
  onToggleBlocks,
  loading,
  unsavedChanges
}) {
  // const currentStyle = editorState.getCurrentInlineStyle()

  return (
    <SubMenu
      active="/notes"
      controls={
        <React.Fragment>
          <Link href="/notes" passHref>
            <BackButton>
              <Carret />
              <span>Back</span>
            </BackButton>
          </Link>
          <Controls>
            <Action onClick={() => onToggleMark('bold')}>
              <Icons.BoldIcon />
            </Action>
            <Action divider onClick={() => onToggleMark('italic')}>
              <Icons.ItalicIcon />
            </Action>
            <Action onClick={() => onToggleBlocks('h2')}>
              <Icons.HeaderBigIcon />
            </Action>
            <Action onClick={() => onToggleBlocks('h3')}>
              <Icons.HeaderSmallIcon />
            </Action>
          </Controls>
          <SaveStatus>
            {!loading ? (
              unsavedChanges ? (
                'Not saved'
              ) : (
                'Saved'
              )
            ) : (
              <React.Fragment>
                Saving
                <LoadingElipsis />
              </React.Fragment>
            )}
          </SaveStatus>
        </React.Fragment>
      }
    />
  )
}

EditorToolbar.propTypes = {
  onToggleMark: PropTypes.func.isRequired,
  onToggleBlocks: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  unsavedChanges: PropTypes.bool.isRequired
}
