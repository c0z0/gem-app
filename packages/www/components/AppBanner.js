import React, { useState, useEffect } from 'react'
import styled, { keyframes } from 'styled-components'
import cookie from 'cookie'

import { P } from './Typography'
import Button from './Button'

const Container = styled.div`
  padding: 16px 16px;
  position: relative;
  display: flex;
  align-items: center;
`

const SlideUp = keyframes`
  from {
    transform: translateY(150%);
    opacity: 0;
  }

  to {
    transform: translateY(0%);
    opacity: 1;
  }
`

const CloseButton = styled.button`
  font-size: 16px;
  color: #484848;
  opacity: 0.5;
  position: absolute;
  top: 2px;
  right: 4px;
  padding: 0;
  transform: rotate(45deg);
`

const Banner = styled.div`
  position: fixed;
  bottom: 16px;
  left: 16px;
  right: 16px;
  z-index: 200;

  box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.1);
  border-radius: 10px;
  background: white;

  animation: ${SlideUp} 0.3s ease-out;
`

export default function AppBanner() {
  const [displayBanner, setDisplayBanner] = useState(false)

  useEffect(() => {
    const bannerDismissed =
      cookie.parse(document.cookie).bannerDismissed || false

    setDisplayBanner(
      navigator.userAgent.toLowerCase().includes('android') && !bannerDismissed
    )
  }, [])

  if (displayBanner)
    return (
      <Banner>
        <Container>
          <P style={{ flex: 1 }}>Download our Android app</P>
          <a href="/download-apk">
            <Button>Download</Button>
          </a>
          <CloseButton
            onClick={() => {
              document.cookie = cookie.serialize('bannerDismissed', true, {
                maxAge: 60 * 60 * 24 * 7 // 1 week
              })

              setDisplayBanner(false)
            }}
          >
            +
          </CloseButton>
        </Container>
      </Banner>
    )
  return null
}
