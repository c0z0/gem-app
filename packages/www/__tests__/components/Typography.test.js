import React from 'react'
import { render, cleanup } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import theme from '../../lib/theme'
import 'jest-styled-components'

import { Title, LoadingElipsis, P, H1 } from '../../components/Typography'

afterEach(cleanup)

test('Title matches snapshot', () => {
  const { container, rerender } = render(
    <ThemeProvider theme={theme}>
      <Title>Test title</Title>
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()

  rerender(
    <ThemeProvider theme={theme}>
      <Title light>Test text</Title>
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()
})

test('P matches snapshot', () => {
  const { container, rerender } = render(
    <ThemeProvider theme={theme}>
      <P>Test text</P>
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()

  rerender(
    <ThemeProvider theme={theme}>
      <P light>Test text</P>
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()
})

test('LoadingElipsis matches snapshot', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <LoadingElipsis />
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()
})

test('H1 matches snapshot', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <H1>Test heading</H1>
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()
})
