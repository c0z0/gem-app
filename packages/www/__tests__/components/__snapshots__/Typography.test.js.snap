// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 matches snapshot 1`] = `
.c0 {
  font-size: 32px;
  margin: 2em 0 1em 0;
  font-weight: 600;
}

@media (max-width:991px) {
  .c0 {
    margin: 1em 0 1em 0;
  }
}

<div>
  <h1
    class="c0"
  >
    Test heading
  </h1>
</div>
`;

exports[`LoadingElipsis matches snapshot 1`] = `
.c0 {
  -webkit-animation: gbsBeW 1.4s infinite;
  animation: gbsBeW 1.4s infinite;
}

.c0:nth-child(2) {
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}

.c0:nth-child(3) {
  -webkit-animation-delay: 0.4s;
  animation-delay: 0.4s;
}

<div>
  <span
    class="c0"
  >
    .
  </span>
  <span
    class="c0"
  >
    .
  </span>
  <span
    class="c0"
  >
    .
  </span>
</div>
`;

exports[`P matches snapshot 1`] = `
.c0 {
  line-height: 1.5rem;
}

<div>
  <p
    class="c0"
  >
    Test text
  </p>
</div>
`;

exports[`P matches snapshot 2`] = `
<div>
  <p
    class="sc-htpNat kDwRwl"
  >
    Test text
  </p>
</div>
`;

exports[`Title matches snapshot 1`] = `
.c0 {
  font-weight: 400;
  color: #75489B;
  font-size: 24px;
  margin: 0;
}

<div>
  <h1
    class="c0"
  >
    Test title
  </h1>
</div>
`;

exports[`Title matches snapshot 2`] = `
<div>
  <h1
    class="sc-bdVaJa jvoJiL"
  >
    Test text
  </h1>
</div>
`;
